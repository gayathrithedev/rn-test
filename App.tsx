/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 */

import React from 'react';
import {SafeAreaView, ScrollView, Text, View, StyleSheet} from 'react-native';

function App(): React.JSX.Element {
  return (
    <SafeAreaView>
      <ScrollView contentInsetAdjustmentBehavior="automatic">
        <View>
          <Text style={styles.title}>Welcome home</Text>
        </View>
      </ScrollView>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  title: {
    fontSize: 20,
  },
});

export default App;
