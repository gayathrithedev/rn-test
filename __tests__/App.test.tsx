/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';

// Note: import explicitly to use the types shipped with jest.
import {it, test, expect} from '@jest/globals';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  renderer.create(<App />);
});

test('should match the tree', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
