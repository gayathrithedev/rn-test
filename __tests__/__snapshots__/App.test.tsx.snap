// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the tree 1`] = `
<RCTSafeAreaView>
  <RCTScrollView
    contentInsetAdjustmentBehavior="automatic"
  >
    <View>
      <View>
        <Text
          style={
            {
              "fontSize": 20,
            }
          }
        >
          Welcome home
        </Text>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
